src/pocketmine/item/WrittenBook.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\item;
+
+use pocketmine\nbt\tag\IntTag;
+use pocketmine\nbt\tag\StringTag;
+
+class WrittenBook extends WritableBook{
+
+	public function __construct(int $meta = 0){
+		Item::__construct(self::WRITTEN_BOOK, $meta, "Written Book");
+	}
+
+	public function getMaxStackSize() : int{
+		return 16;
+	}
+
+	/**
+	 * Returns the generation of the book.
+	 * Generations higher than 1 can not be copied.
+	 *
+	 * @return int
+	 */
+	public function getGeneration() : int{
+		if(!isset($this->getNamedTag()->generation)) {
+			return -1;
+		}
+		return $this->getNamedTag()->generation->getValue();
+	}
+
+	/**
+	 * Sets the generation of a book.
+	 *
+	 * @param int $generation
+	 */
+	public function setGeneration(int $generation) : void{
+		if($generation < 0 or $generation > 3){
+			throw new \InvalidArgumentException("Generation \"$generation\" is out of range");
+		}
+		$namedTag = $this->getCorrectedNamedTag();
+
+		if(isset($namedTag->generation)){
+			$namedTag->generation->setValue($generation);
+		}else{
+			$namedTag->generation = new IntTag("generation", $generation);
+		}
+		$this->setNamedTag($namedTag);
+	}
+
+	/**
+	 * Returns the author of this book.
+	 * This is not a reliable way to get the real author of the book. It can be changed in-game.
```

Comment:
is this still applicable?

Reply:
I'm not sure I understand you... written books cannot be modified after being signed.

Reply:
During the process of signing a book, the player can give the book a name, and a PE only feature is that you can enter any name you want as the signer of the book.

Reply:
you're missing my point... a _written_ book cannot be modified. Therefore this is a "reliable" method in terms of "it won't change", however you haven't specified that "this can be anything, not just a player name". The documentation is too ambiguous.

Reply:
Got it, see latest commit.